File: Data Structures & Algorithms/two-integer-sum/submission-0.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        map<int,int> check_dict;
        for (int i = 0; i < nums.size(); i++){
            int remainder = target - nums[i];
            if(check_dict.find(remainder)!=check_dict.end()){
                return{check_dict[remainder],i};
            }
            check_dict[nums[i]] = i;
        }
        return {};
    }
};

File: Data Structures & Algorithms/binary-search/submission-0.py
class Solution:
    def search(self, nums: List[int], target: int) -> int:
        l = 0
        r = len(nums)-1
        while(l<=r):
            mid = l + (r-l)//2  # 0 + (5-0)//2 -> 2
            if(nums[mid]>target): # 2 > 4
                r = mid - 1
            elif(nums[mid]<target): # 2 < 4
                l = mid + 1
            else:
                return mid
        return -1